cc/mcts_player.cc

```diff
@@ -232,18 +232,40 @@ void MctsPlayer::TreeSearch() {
 
 void MctsPlayer::SelectLeaves(MctsNode* root, int num_leaves,
                               std::vector<MctsPlayer::TreePath>* paths) {
-  int max_iterations = num_leaves * 2;
+  DualNet::Output cached_output;
+
+  int max_cache_misses = num_leaves * 2;
   int num_selected = 0;
-  for (int i = 0; i < max_iterations; ++i) {
+  int num_cache_misses = 0;
+  int num_cache_hits = 0;
+  while (num_cache_misses < max_cache_misses) {
     auto* leaf = root->SelectLeaf();
     if (leaf->game_over() || leaf->at_move_limit()) {
       float value =
           leaf->position.CalculateScore(game_->options().komi) > 0 ? 1 : -1;
       leaf->IncorporateEndGameResult(value, root);
+      ++num_cache_misses;
```

Comment:
this appears to be double counted

Reply:
also shouldn't there be a continue here?

Reply:
The rest of the loop is in the else branch so it won't get executed. I've restructured things a bit to make this clearer.

(this bit of code is one of the things I would like to make clearer)